prov/util/src/util_av.c

```diff
@@ -552,7 +552,9 @@ int ofi_av_init_lightweight(struct util_domain *domain, const struct fi_av_attr
 	 */
 	av->context = context;
 	av->domain = domain;
-	ofi_mutex_init(&av->ep_list_lock);
+	ofi_genlock_init(&av->ep_list_lock,
+			 domain->threading != FI_THREAD_SAFE ?
+			       OFI_LOCK_NOOP : OFI_LOCK_MUTEX);
```

Comment:
We can only disable locks for FI_THREAD_DOMAIN.

Reply:
done